### PR TITLE
RDM-13009

### PR DESCRIPTION
**Pull request by KleoG**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-13009

### Change description ###
Update Definition Store to recognise new column - DefaultValue - in the CaseEventToFields tab

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ void testMapEventCaseFieldEntity() {
             eventCaseFieldEntity.setRetainHiddenValue(true);
             eventCaseFieldEntity.setPublish(true);
             eventCaseFieldEntity.setPublishAs("PublishAs test");
+            eventCaseFieldEntity.setDefaultValue("");
```
Comment (danlysiak) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java:132]: Worth setting this to some non-empty string value to prove it is mapping correctly
```suggestion
            eventCaseFieldEntity.setDefaultValue("DefaultValue test");
```

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventCaseFieldParserTest.java [hunk 2]
```diff
@@ -67,10 +67,11 @@ public void defDataItemHasValidShowConditionParseEveCaseFldCalledParsedEventEnti
         String originalShowCondition = "Original Show Condition";
         String label = "label";
         String hint = "hint";
+        String defaultValue = "";
```
Comment (danlysiak) [inline excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventCaseFieldParserTest.java:70]: Similarly here, can we add a non-empty value please

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java [hunk 3]
```diff
@@ -153,6 +154,19 @@ public void setPublish(Boolean publish) {
         this.publish = publish;
     }
 
+    /**
+     * whenever we want to publish this event field.
+     **/
```
Comment (danlysiak) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java:159]:
```suggestion
```
Copied but not changed from other `publish` - probably not any benefit to having a comment for these field anyway, when there is a description in the line below, so could just delete

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java [hunk 4]
```diff
@@ -153,6 +154,19 @@ public void setPublish(Boolean publish) {
...
+    @ApiModelProperty(value = "allows services to define the default value of the CaseAccessCategory")
```
Comment (danlysiak) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java:160]:
```suggestion
    @ApiModelProperty(value = "allows services to define the default value of top level case fields")
```
`CaseAccessCategory` is the first usage, but in theory can be used for anything

File: aat/src/aat/resources/features/F-103 New DefaultValue Column/F-103.feature [hunk 5]
```diff
@@ -0,0 +1,23 @@
+@F-103
+Feature: F-103: New DefaultValue Column
```
Comment (danlysiak) [inline aat/src/aat/resources/features/F-103 New DefaultValue Column/F-103.feature:2]: Missing scenario for when there **is** a `DefaultValue`?

File: aat/src/aat/resources/features/F-103 New DefaultValue Column/S-103.2.td.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	"specs": [
+		"an active profile in CCD",
+		"contains a missing default value column"
+	],
+  "request": {
+    "body": {
+      "arrayInMap": [
+        {
+          "key": "file",
+          "filePath": "uk/gov/hmcts/ccd/test_definitions/invalid/BEFTA_Master_Definition_Missing_CaseEventToFields_DefaultValue_Column.xlsx"
```
Comment (danlysiak) [inline aat/src/aat/resources/features/F-103 New DefaultValue Column/S-103.2.td.json:14]: This path doesn't sound quite correct - I thought the `invalid` definitions are ones where we expect validation errors. Happy to be corrected though

**Review by danlysiak — COMMENTED** [review]
A few minor comments; implementation-wise looks good to me

File: aat/src/aat/resources/features/F-103 New DefaultValue Column/S-103.2.td.json (hunk 6 above)
Comment (KleoG) [reply aat/src/aat/resources/features/F-103 New DefaultValue Column/S-103.2.td.json]: Fixed in: [d439d7](https://github.com/hmcts/ccd-definition-store-api/pull/1111/commits/d439d7655d27627a95440edc5b0fc6a30fd3cab8)

File: aat/src/aat/resources/features/F-103 New DefaultValue Column/F-103.feature (hunk 5 above)
Comment (KleoG) [reply aat/src/aat/resources/features/F-103 New DefaultValue Column/F-103.feature:2]: Fixed in [5e6bcf3](https://github.com/hmcts/ccd-definition-store-api/pull/1111/commits/5e6bcf3530b16e5bff37a8f0c02855a0d686d834)

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java (hunk 4 above)
Comment (KleoG) [reply repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java]: Fixed in [5e6bcf3](https://github.com/hmcts/ccd-definition-store-api/pull/1111/commits/5e6bcf3530b16e5bff37a8f0c02855a0d686d834)

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventCaseFieldParserTest.java (hunk 2 above)
Comment (KleoG) [reply excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventCaseFieldParserTest.java]: Fixed in [5e6bcf3](https://github.com/hmcts/ccd-definition-store-api/pull/1111/commits/5e6bcf3530b16e5bff37a8f0c02855a0d686d834)

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java (hunk 1 above)
Comment (KleoG) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java]: Fixed in [5e6bcf3](https://github.com/hmcts/ccd-definition-store-api/pull/1111/commits/5e6bcf3530b16e5bff37a8f0c02855a0d686d834)

File: build.gradle [hunk 7]
```diff
@@ -269,7 +269,7 @@ subprojects { subproject ->
         testCompile "io.rest-assured:rest-assured:${restAssuredVersion}"
         testCompile group: 'org.powermock', name: 'powermock-api-mockito2', version: powermockVersion
         testCompile group: 'org.powermock', name: 'powermock-module-junit4', version: powermockVersion
-        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.9.1'
+        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.9.9.2-prerelease'
```
Comment (mattnayler) [inline build.gradle:272]: This needs to be updated to a full release version of the hmcts/ccd-test-defintions before going into master. see hmcts/ccd-test-definitions#35.

**Review by mattnayler — APPROVED** [review]
👍  LGTM, subject to the usual update of the `ccd-test-definitions` version once a new release of that is available.

**Review by mattnayler — APPROVED** [review]
Still LGTM after late changes to FTAs.  thanks

File: build.gradle [hunk 8]
```diff
@@ -269,7 +269,7 @@ subprojects { subproject ->
         testCompile "io.rest-assured:rest-assured:${restAssuredVersion}"
         testCompile group: 'org.powermock', name: 'powermock-api-mockito2', version: powermockVersion
         testCompile group: 'org.powermock', name: 'powermock-module-junit4', version: powermockVersion
-        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.9.1'
+        testCompile group: 'com.github.hmcts', name: 'ccd-test-definitions', version: '7.10.1-prerelease'
```
Comment (mattnayler) [inline build.gradle:272]: This needs to be updated to a full release version of the hmcts/ccd-test-defintions before going into master. see https://github.com/hmcts/ccd-test-definitions/pull/35.

**Pull request merged**